doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1352 @@
+
+```
+1. show vlan id and brief will be enhanced to display DIP. (tunnel members from  APP DB)
+
+2. show mac /show mac -v will be enhanced for EVPN MACs.
+   - Port column will display DIP. (ASIC DB view)
+   - Type column will display EVPN_static/EVPN_dynamic
+
+3. show NeighbourSuppressStatus to display Neighbor suppression status.
+
+```
+
+#### 5.2.3 Validations 
+
+```
+- VLAN to be configured before map vlan vni command is executed. 
+- VLAN cannot be deleted if there is a vlan vni map. 
+- source to be configured before  map vlan vni command is executed.
+- source cannot be removed if there are mappings already present. 
+- interface vxlan cannot be deleted if evpn nvo is configured. 
+- VRF is already configured    
+- VNI VLAN map should be configured prior to VRF VNI map configuration, since VNI should be both L2 and L3.
+- VNI VLAN map cannot be deleted if there is corresponding VRF VNI map.
+
+```
+
+## 6 Serviceability and Debug
+
+The existing logging mechanisms shall be used. The show commands and the redis table
+dumps can be used as debug aids.
+
+## 7 Warm Reboot Support
+
+Warm reboot will support below requirements
+
+1. System Warm reboot for EVPN 
+   The goal of the system warm reboot is to achieve system software reload without impacting the data plane and graceful restart of the control plane (BGP).
+
+![System Warm reboot and reconciliation sequence](images/warmrebootseq.PNG "Figure 16: System Warm reboot and reconciliation sequence")
+
+__Figure 16: System Warm reboot and reconciliation sequence__
+
+During the system warm-reboot restoration/reconciliation, the fdbsyncd is dependent on Intf, Vlan, Vrf, Vxlan Mgrs to complete their data restoration to kernel. Before fdbsyncd can program the MACs to the kernel it needs to wait for these subsystems data replay to kernel. Without this the kernel programming for fdbsyncd will fail.  Subsequently if the FDB/VTEP info is not available in kernel/zebra for BGP to distribute to its peers, it will converge with premature info causing data-loss. To solve this dependency , we use a replay-done notification by each subsystem. The notification is passed between the subsystems by using the existing  WARM_RESTART_TABLE. A new state "replayed" in the data restoration flow  ( "state" = "replayed") is used by each subsystem to indicate it has replayed its data to kernel.
+
+The dependent subsystems use the WARM_RESTART_TABLE table to wait before attempting to program their info to the kernel.   fdbsyncd replays its info to kernel only after the data is replayed for Intf, Vlan, Vxlan and Vrf Mgrs.  BGP is configured/started only when FDB has replayed all its data to kernel. 
+
+There is also a requirement that the app_db entries are not modified by *syncd subsystems until the orchagent has reconciled its data after warm-reboot. To achieve this dependency, the *syncd applications which write data to the app_db wait till orchagent reports "reconciled" status in the WARM_RESTART_TABLE.
```

Comment:
>There is also a requirement that the app_db entries are not modified by *syncd subsystems until the orchagent has reconciled its data after warm-reboot [](start = 0, length = 151)

I think there is no such requirement. Could you check?

Reply:
This mechanism has been put in place in this design to ensure that the orchagent/syncd/SAI restores the s/w states, h/w states to "pre warmboot states".
The "delta" changes during the warmboot process exchanged from all the remote EVPN peers are replayed later as an update to ensure that the syncd/SAI updates to the latest available state.
In this process, the following things happen in parallel.
After all the process like vlan/vxlan/intf/vrf has replayed the existing config and reconciled with kernel/app_db, the FRR/BGP starts the process of communicating with the remote peers to send/receive the latest underlay/overlay mac/arp/route information.
During the same time, the orchagent/syncd/SAI, tries to restore the "pre-warmboot" state based on the app/asic DB.
Once both the above process are finished (one may finish before the other), if there is any delta changes, they are replayed and reprogram the hardware.
With the above design approach, there is a better control and deterministic behavior.
Please let us know if you have any further questions.